R/xgb_binning.R

```diff
@@ -0,0 +1,250 @@
+#' Discretize numeric variables with XgBoost
+#'
+#' `step_xgb_binning` creates a *specification* of a recipe
+#'  step that will discretize numeric data (e.g. integeres or doubles)
+#'  into bins in a supervised way using an XgBoost model.
+#'
+#' @param recipe A recipe object. The step will be added to the sequence of
+#'   operations for this recipe.
+#' @param ... One or more selector functions to choose which variables are
+#'   affected by the step. See [selections()] for more details.
+#' @param role Defaults to `"predictor"`.
+#' @param trained A logical to indicate if the quantities for preprocessing
+#'   have been estimated.
+#' @param outcome A character string to indicate the outcome variable to train XbBoost models.
```

Comment:
This is currently inconsistent with some `recipes` steps, can you use tidy select here? See tidymodels/embed#23

Reply:
As noted below, this could also work for regression problems so maybe consider relaxing this a bit.

Reply:
Ok, so you mean to define the outcome in a similar way as it's currently in this example:

```
embed::step_lencode_bayes(pregnant_fct, outcome = vars(diabetes))
```

Reply:
Yes.

Reply:
> Ok, done.